### PR TITLE
SNOW-1727330, SNOW-1727332, SNOW-1727334, SNOW-1727335: Add support for DataFrameGroupBy/SeriesGroupBy.bfill/ffill

### PR DESCRIPTION
**Pull request by sfc-gh-helmeleegy**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1727330, SNOW-1727332, SNOW-1727334, SNOW-1727335

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.
   - [ ] I acknowledge that I have ensured my changes to be thread-safe. Follow the link for more information: [Thread-safe Developer Guidelines](https://docs.google.com/document/d/162d_i4zZ2AfcGRXojj0jByt8EUq-DrSHPPnTa4QvwbA/edit#bookmark=id.e82u4nekq80k)

3. Please describe how your code solves the related issue.

   Add support for DataFrameGroupBy/SeriesGroupBy.bfill/ffill.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -60,6 +60,7 @@
 - Added support for applying Snowpark Python functions (e.g., `sin`) in `Series.map`, `Series.apply`, `DataFrame.apply` and `DataFrame.applymap`.
 - Added support for `np.subtract`, `np.multiply`, `np.divide`, and `np.true_divide`.
 - Added support for tracking usages of `__array_ufunc__`.
+- Added support for `DataFrameGroupBy.bfill`, `SeriesGroupBy.bfill`, `DataFrameGroupBy.ffill`, and `SeriesGroupBy.ffill`.
```
Comment (sfc-gh-nkrishna) [inline CHANGELOG.md:63]: Can you move this to 1.24.0 Snowpark pandas new features section?

**Review by sfc-gh-nkrishna — APPROVED** [review]
LGTM, just one comment around fixing the CHANGELOG

File: CHANGELOG.md (hunk 1 above)
Comment (sfc-gh-jjiao) [reply CHANGELOG.md:63]: +1

Comment (sfc-gh-helmeleegy) [reply CHANGELOG.md]: Done.

**Review by sfc-gh-joshi — APPROVED** [review]
LGTM, just one question: did we already support `DataFrameGroupBy.fillna` with both `method="ffill"` and `method="bfill"`, but we just didn't expose those methods through `DataFrameGroupBy.bfill` and `DataFrameGroupBy.ffill`?

**Comment by sfc-gh-helmeleegy** [comment]
> LGTM, just one question: did we already support `DataFrameGroupBy.fillna` with both `method="ffill"` and `method="bfill"`, but we just didn't expose those methods through `DataFrameGroupBy.bfill` and `DataFrameGroupBy.ffill`?

Yes, `DataFrameGroupBy.fillna` was already supported.

**Pull request merged**